src/dbnode/storage/namespace.go

```diff
@@ -1602,3 +1604,171 @@ func (n *dbNamespace) FlushState(shardID uint32, blockStart time.Time) (fileOpSt
+
+	n.RLock()
+	if n.bootstrapState != Bootstrapped {
+		n.RUnlock()
+		return 0, errNamespaceNotBootstrapped
+	}
+	nsCtx := n.nsContextWithRLock()
+	n.RUnlock()
+
+	targetShards := n.OwnedShards()
+
+	// Note: Cold writes must be enabled for Large Tiles to work.
+	if !n.nopts.ColdWritesEnabled() {
+		return 0, errColdWritesDisabled
+	}
+
+	sourceBlockSize := sourceNs.Metadata().Options().RetentionOptions().BlockSize()
+	sourceBlockStart := opts.Start.Truncate(sourceBlockSize)
+
+	sourceNsOpts := sourceNs.StorageOptions()
+	reader, err := fs.NewReader(sourceNsOpts.BytesPool(), sourceNsOpts.CommitLogOptions().FilesystemOptions())
+	if err != nil {
+		return 0, err
+	}
+
+	wOpts := series.WriteOptions{
+		TruncateType: n.opts.TruncateType(),
+		SchemaDesc:   nsCtx.Schema,
+	}
+
+	resources, err := newColdFlushReuseableResources(n.opts)
+	if err != nil {
+		return 0, err
+	}
+
+	// NB(bodu): Deferred targetShard cold flushes so that we can ensure that cold flush index data is
+	// persisted before persisting TSDB data to ensure crash consistency.
+	multiErr := xerrors.NewMultiError()
+	var processedBlockCount int64
+	for _, targetShard := range targetShards {
+		sourceShard, _, err := sourceNs.readableShardAt(targetShard.ID())
+		if err != nil {
+			detailedErr := fmt.Errorf("no matching shard in source namespace %s: %v", sourceNs.ID(), err)
+			multiErr = multiErr.Add(detailedErr)
+			continue
+		}
+		shardProcessedBlockCount, err := targetShard.AggregateTiles(ctx, reader, sourceNs.ID(), sourceBlockStart, sourceShard, opts, wOpts)
+		processedBlockCount += shardProcessedBlockCount
+		if err != nil {
+			detailedErr := fmt.Errorf("shard %d aggregation failed: %v", targetShard.ID(), err)
+			multiErr = multiErr.Add(detailedErr)
+			continue
+		}
+
+		multiErr = n.coldFlushSingleShard(nsCtx, targetShard, pm, resources, multiErr)
+	}
+
+	return processedBlockCount, multiErr.FinalError()
+}
+
+func (n *dbNamespace) coldFlushSingleShard(
+	nsCtx namespace.Context,
+	shard databaseShard,
+	pm persist.Manager,
+	resources coldFlushReuseableResources,
+	multiErr xerrors.MultiError,
+) xerrors.MultiError {
+	// NB(rartoul): This value can be used for emitting metrics, but should not be used
+	// for business logic.
+	callStart := n.nowFn()
+
+	// NB(bodu): The in-mem index will lag behind the TSDB in terms of new series writes. For a period of
+	// time between when we rotate out the active cold mutable index segments (happens here) and when
+	// we actually cold flush the data to disk we will be making writes to the newly active mutable seg.
+	// This means that some series can live doubly in-mem and loaded from disk until the next cold flush
+	// where they will be evicted from the in-mem index.
+	var (
+		onColdFlushDone OnColdFlushDone
+		err             error
+	)
+	if n.reverseIndex != nil {
+		onColdFlushDone, err = n.reverseIndex.ColdFlush([]databaseShard{shard})
+		if err != nil {
+			n.metrics.writeAggregated.ReportError(n.nowFn().Sub(callStart))
```

Comment:
Remove all these since they duplicate the top level n.metrics.writeAggregated.ReportErrorOrSuccess(...) call from `AggregateTiles(...)`

Reply:
In latest version there's just a single `ReportSuccessOrError` (as the cold flush is no longer in use).